File: JAVA + DSA Program/RECURSION/factorial.java
public class factorial {

    public static void main(String[] args) {
        int n = 5;
        int result = factorials(n);
        System.out.println("Factorial of " + n + " is: " + result);
        // Output: Factorial of 5 is: 120
    }
    public static int factorials(int n){
        if(n == 0){
            return 1;
        }
        return n * factorials(n -1);

    }    
}

File: CPROGRAM.c/FUNCTION.C/Factorial.c
#include <stdio.h>

// Function to calculate factorial using recursion
int factorial(int n) {
    // Base case: factorial of 0 or 1 is 1
    if (n == 0 || n == 1) {
        return 1;
    }
    // Recursive case: n! = n * (n-1)!
    else {
        return n * factorial(n - 1);
    }
}

int main() {
    int num;
    printf("Enter a number: ");
    scanf("%d", &num);

    // Check if the number is negative
    if (num < 0) {
        printf("Error: Factorial is not defined for negative numbers.\n");
    }
    else {
        int result = factorial(num);
        printf("Factorial of %d = %d\n", num, result);
    }

    return 0;
}